### PR TITLE
Fix bug in references with none

### PR DESCRIPTION
**Pull request by elronbandel**

### CODE REVIEW

File: src/unitxt/operators.py [hunk 1]
```diff
@@ -1480,6 +1480,11 @@ class RemoveValues(FieldOperator):
 
     def verify(self):
         super().verify()
+        if self.process_every_value:
```
Comment (dafnapension) [inline src/unitxt/operators.py:1483]: why is this needed?
the operator operates on whats given it it. The operator just needs to verify that it is given a list and not a scalar.  Why should it care of the source of what is given to it? whether it was every value of a parent list or not?

Comment (elronbandel) [reply src/unitxt/operators.py:1483]: I dont know, but it does not part of this PR its part of the reverting of your last commit. (to fix the bugs in the most minimal way) we can do this in a different commit.
